correct_labels.py

```diff
@@ -0,0 +1,184 @@
+            self.save_path = labels_path
+        else:
+            self.labels = zarr.open(labels_path, mode='r+')[frame]
+            # Get the save path 
+            data_path = Path(labels_path)
+            self.save_path = os.path.join(data_path.parents[0], 
+                                          data_path.stem + f'_t{frame}_GT.zarr')
+
+
+    def __call__(self):
+        with napari.gui_qt():
+            v = napari.Viewer()
+            print(type(self.labels))
+            v.add_image(self.array, name='Image', scale=self.scale)
+            v.add_labels(self.labels, name='Labels', scale=self.scale)
+            v.add_points(np.empty((0, len(self.labels.shape)), dtype=float), scale=self.scale, size=2)
+            v.bind_key('1', self._points)
+            v.bind_key('2', self._watershed)
+            v.bind_key('3', self._select_colour)
+            v.bind_key('4', self._fill)
+            v.bind_key('5', self._paint)
+            v.bind_key('Shift-s', self._save)
+    
+
+    def _points(self, viewer):
+        """
+        Switch to points layer to split a label
+        """
+        if viewer.layers['Points'].mode != 'add':  
+            viewer.layers['Points'].mode = 'add' 
+        else:   
+            viewer.layers['Points'].mode = 'pan_zoom' 
+
+
+    def _watershed(self, viewer):
+        """
+        Execute watershed to split labels based on provided points. 
+        Executes over one frame. 
+        """
+        # find the labels corresponding to the current points in the points layer
+        labels = viewer.layers['Labels'].data
+        image = viewer.layers['Image'].data
+        points = viewer.layers['Points'].data
+        points = np.round(points).astype(int)
+        labels = watershed_split(image, labels, points)
+        viewer.layers['Labels'].data = labels
+        viewer.layers['Points'].data = np.empty((0, points.shape[1]), dtype=float)
+        viewer.layers['Labels'].refresh()    
+    
+
+    def _select_colour(self, viewer):
+        """
+        Select colour for painting
+        """
+        if viewer.layers['Labels'].mode != 'pick': 
+            viewer.layers['Labels'].mode = 'pick'  
+        else:
+            viewer.layers['Labels'].mode = 'pan_zoom'  
+    
+
+    def _fill(self, viewer):
+        """
+        Switch napari labels layer to fill mode
+        """
+        if viewer.layers['Labels'].mode != 'fill':
+            viewer.layers['Labels'].mode = 'fill' 
+        else:
+            viewer.layers['Labels'].mode = 'pan_zoom'
+
+
+    def _paint(self, viewer):
+        """
+        Switch napari labels layer to paint mode
+        """
+        if viewer.layers['Labels'].mode != 'paint':  
+            viewer.layers['Labels'].mode = 'paint'  
+        else:  
+            viewer.layers['Labels'].mode = 'pan_zoom'
+    
+
+    def _save(self, viewer): 
+        # this damn thing that made me write this into a class :facepalm:
+        # Perhaps there is a better way?
```

Comment:
Haha I think so! This is because you need self.save_path? You can use a curried function:

```python
import toolz as tz

@tz.curry
def _save(viewer, path):
    output = viewer.layers['Labels'].data
    zarr.save_array(path, output)

...

v.bind_key('Shift-s', _save(path=save_path))
```

Reply:
Oh yeah, I forgot  🤦‍♀️ 🤣